#2128 - Fix device mapper partitioning

tlbueno opened the pull request:
When run ignition on a device mapper, ie, multipath, it fails because the function blockDevHeld returns true as the block device contains holders. A block device with holders do not necessary means the block device is in use (like mounted).
The function blockDevInUse should look for mounted devices only.

fixes #2128

prestist commented on internal/exec/stages/disks/partitions.go:
```diff
@@ -323,18 +323,6 @@ func (p PartitionList) Swap(i, j int) {
-
-	holdersDir := fmt.Sprintf("/sys/class/block/%s/holders/", blockDevNode)
-	entries, err := os.ReadDir(holdersDir)
-	if err != nil {
```
Hmm, I am a little worried about removing this check on holder's I have to imagine there are edge cases that would need us to check. Lets flip this change so we make a more direct alteration when doing multi path explicitly.

tlbueno committed:
Fix device mapper partitioning

When run ignition on a device mapper, ie, multipath, it fails because
the function blockDevHeld returns true as the block device
contains holders. A block device with holders do not necessary means
the block device is in use (like mounted).
The function blockDevInUse will not check if it is a device mapper
and if so, do not check for blockDevHeld.

Signed-off-by: Tiago Bueno <tiago.bueno@gmail.com>

prestist reviewed (commented):
Overall lgtm, have you had a change to verify that this fixes the issue?

I am wary of updating the partitioning logic checks simply because these changes will effect all specs as well, but there are a lot of partitioning tests so that makes me feel slightly better.

tlbueno commented:
> Overall lgtm, have you had a change to verify that this fixes the issue?
> 
> I am wary of updating the partitioning logic checks simply because these changes will effect all specs as well, but there are a lot of partitioning tests so that makes me feel slightly better.

yes, with that change the issue is fixed.

tlbueno commented on docs/release-notes.md:
```diff
@@ -28,6 +28,7 @@ nav_order: 9
 ### Bug fixes
 
 - Fix fetch-offline for Oracle Cloud Infrastructure
+- Fix device mapper partitioning ([#2128](https://github.com/coreos/ignition/issues/2128))
```
yeah...nice catch. I am going to update it

jlebon reviewed (approved):
One comment re. release notes, but LGTM otherwise.

I think we could've kept the holders check but just make it smarter to detect the case where it's really just partition devices and not a holder that claims the whole device. But I'm not sure it's worth it right now.
